src/main/java/com/google/devtools/build/lib/packages/Attribute.java

```diff
@@ -287,44 +286,6 @@ public String checkValid(Rule from, Rule to) {
         }
       };
 
-  /**
-   * Provides a {@link SplitTransition} given the originating target {@link Rule}. The split
-   * transition may be constant for all instances of the originating rule, or it may differ
-   * based on attributes of that rule. For instance, a split transition on a rule's deps may differ
-   * depending on the 'platform' attribute of the rule.
-   */
-  public interface SplitTransitionProvider extends SplitTransitionProviderApi {
-    /**
-     * Returns the {@link SplitTransition} given the attribute mapper of the originating rule.
-     */
-    SplitTransition apply(AttributeMap attributeMap);
-  }
-
-  /**
-   * Implementation of {@link SplitTransitionProvider} that returns a single {@link SplitTransition}
-   * regardless of the originating rule.
-   */
-  @AutoCodec.VisibleForSerialization
-  @AutoCodec
-  static class BasicSplitTransitionProvider implements SplitTransitionProvider {
-    private final SplitTransition splitTransition;
-
-    @AutoCodec.VisibleForSerialization
-    BasicSplitTransitionProvider(SplitTransition splitTransition) {
-      this.splitTransition = splitTransition;
-    }
-
-    @Override
-    public SplitTransition apply(AttributeMap attributeMap) {
-      return splitTransition;
-    }
-
-    @Override
-    public void repr(SkylarkPrinter printer) {
-      printer.append("<transition object>");
```

Comment:
How do these get printed in Starlark with this method removed?

Reply:
Only starlark transitions need to be printed from Starlark, and they all implement SplitTransitionProviderApi (and so have a repr method).